File: persistence/src/main/java/com/pij/zworkout/persistence/xml/DummyWorkoutSerializerService.java
package com.pij.zworkout.persistence.xml;

import com.pij.zworkout.persistence.api.PersistableWorkout;
import com.pij.zworkout.persistence.api.WorkoutSerializerService;

import java.io.OutputStream;

import io.reactivex.Completable;

/**
 * <p>Created on 13/03/2018.</p>
 *
 * @author Pierrejean
 */
class DummyWorkoutSerializerService implements WorkoutSerializerService {

    @Override
    public Completable write(PersistableWorkout data, OutputStream target) {
        return Completable.error(new UnsupportedOperationException("write([data, target]) not implemented."));
    }

}

File: persistence/src/main/kotlin/com/pij/zworkout/persistence/xml/DummyWorkoutSerializerService.kt
package com.pij.zworkout.persistence.xml

import com.pij.zworkout.persistence.api.PersistableWorkout
import com.pij.zworkout.persistence.api.WorkoutSerializerService
import io.reactivex.Completable
import io.reactivex.Single
import java.io.InputStream
import java.io.OutputStream

/**
 *
 * Created on 13/03/2018.
 *
 * @author Pierrejean
 */
internal class DummyWorkoutSerializerService : WorkoutSerializerService {

    override fun write(data: PersistableWorkout, target: OutputStream): Completable {
        return Completable.error(UnsupportedOperationException("write([data, target]) not implemented."))
    }

    override fun read(source: InputStream): Single<PersistableWorkout> {
        return Single.error(UnsupportedOperationException("read([source, data]) not implemented."))
    }

}
